tensorflow/python/ops/ragged/ragged_embedding_ops.py

```diff
@@ -0,0 +1,242 @@
+
+  with ops.name_scope(name, "embedding_lookup_ragged") as name:
+    looked_up_ragged = ragged_functional_ops.map_flat_values(
+        embedding_ops.embedding_lookup,
+        params=params,
+        ids=ids,
+        partition_strategy=partition_strategy,
+        max_norm=max_norm)
+
+    return looked_up_ragged
+
+@dispatch.dispatch_for_api(embedding_ops.embedding_lookup_sparse)
+def embedding_lookup_sparse(params,
+                            sp_ids: ragged_tensor.Ragged,
+                            sp_weights,
+                            partition_strategy="mod",
+                            name=None,
+                            combiner=None,
+                            max_norm=None,
+                            allow_dense_grads=False):
+  if combiner is None:
+    combiner = "mean"
+  if combiner not in ("mean", "sqrtn", "sum"):
+    raise ValueError(
+        f"combiner must be one of 'mean', 'sqrtn' or 'sum', got {combiner}")
+  if isinstance(params, variables.PartitionedVariable):
+    params = list(params)  # Iterate to get the underlying Variables.
+  if not isinstance(params, list):
+    params = [params]
+  ignore_weights = sp_weights is None
+  if not ignore_weights:
+    if not isinstance(sp_weights, ragged_tensor.RaggedTensor):
+      raise TypeError(f"sp_ids must be of the same type as sp_weights.")
+    sp_ids.values.get_shape().assert_is_compatible_with(
+        sp_weights.values.get_shape())
+    sp_ids.get_shape().assert_is_compatible_with(
+        sp_weights.get_shape())
+    # TODO(yleon): Add enhanced node assertions to verify that sp_ids and
+    # sp_weights have equal indices and shapes.
+
+  with ops.name_scope(name, "embedding_lookup_sparse",
+                      params + [sp_ids]) as name:
+
+    segment_ids = sp_ids.value_rowids()
+    ids = sp_ids.flat_values
+
+    if len(params) == 1 and max_norm is None and allow_dense_grads:
+      idx = ids
+      embeddings = params[0]
+    else:
+      ids, idx = array_ops.unique(ids)
+      embeddings = embedding_ops.embedding_lookup(
+          params, ids, partition_strategy=partition_strategy, max_norm=max_norm)
+
+    return embedding_ops.embedding_lookup_sparse_impl(embeddings, segment_ids,
+                                                      sp_weights, idx, combiner,
+                                                      ignore_weights, name)
+
+
+@dispatch.dispatch_for_api(embedding_ops.safe_embedding_lookup_sparse)
+def safe_embedding_lookup_sparse(embedding_weights,
+                                 sparse_ids: ragged_tensor.Ragged,
+                                 sparse_weights=None,
+                                 combiner="mean",
+                                 default_id=None,
+                                 name=None,
+                                 partition_strategy="div",
+                                 max_norm=None,
+                                 allow_dense_grads=False):
```

Comment:
Also add a docstring here? In the future, we want to use this docstring as the offical docstring for RaggedTensor.